app/Models/Photo.php

```diff
@@ -438,17 +399,13 @@ public function toArray(): array
 			$result['is_public'] = $result['is_public'] === true ? 1 : 0;
 		}
 
+		$result['rights'] = PhotoRights::ofPhoto($this);
+
 		// Downgrades the accessible resolution of a photo
-		// The decision logic here is a merge of three formerly independent
-		// (and slightly different) approaches
 		if (
-			!Gate::check(PhotoPolicy::IS_OWNER, $this) &&
+			!$result['rights']->can_access_full_photo &&
```

Comment:
```suggestion
			!Gate::check(PhotoPolicy::CAN_ACCESS_FULL_PHOTO, [Photo::class, $this]) &&
```

because `$result['rights']` should not be added to the result (see above).